lib/sheets_db/worksheet.rb

```diff
@@ -45,17 +45,24 @@ def attribute_definitions
     end
 
     def get_definition_and_column(attribute_name)
-      attribute_definition = attribute_definitions.fetch(attribute_name, {})
+      attribute_definition = attribute_definitions.fetch(attribute_name)
       column_name = attribute_definition.fetch(:column_name, attribute_name.to_s)
-      raise ColumnNotFoundError, column_name if columns[column_name].nil?
       [
         attribute_definition,
         columns[column_name]
       ]
     end
 
+    def value_if_column_missing(attribute_definition)
+      unless attribute_definition[:if_column_missing]
+        raise ColumnNotFoundError, attribute_definition[:column_name]
```

Comment:
Can we collect the `ColumnNotFoundErrors` in case there is more than one and raise with an array of the missing columns? That way the user won't have re-run ingestion to find more errors. Not a blocker, but would be great to have.

Reply:
In this specific instance, that doesn't make sense, since it's raised while querying for a value - basically a "no method error" sort of thing.  But having a `#validate` method on a worksheet is a great idea - something you can explicitly run or that's run on instantiation.  That method would run through all the column headers and the attribute definitions and raise an exception if there were any required defined columns that were missing.  Something to look into later.